core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/NetworkCommunicationActor.scala

```diff
@@ -174,6 +338,11 @@ class NetworkCommunicationActor(parentRef: ActorRef, val actorId: ActorVirtualId
       if (parentRef != null) {
         fetchActorRefMappingFromParent(actorID)
       }
+    case PollForCredit(to) =>
+      val req = NetworkMessage(networkMessageID, CreditRequest(actorId))
+      messageIDToIdentity(networkMessageID) = to
```

Comment:
The polling request is adding an entry to the map every 200ms. This may be a memory concern. A solution is to create a separate type of message for polling that we don't need to log and the ack from that is handled separately.